controllers/cloud.redhat.com/providers/kafka/appinterface.go

```diff
@@ -92,11 +95,12 @@ func validateBrokerService(ctx context.Context, cl client.Client, nn types.Names
 	err := cl.Get(ctx, nn, &svc)
 
 	if err != nil {
-		errors.LogError(ctx, "kafka", errors.New("Cannot find kafka bootstrap service"))
-		qualifiedName := fmt.Sprintf("%s:%s", nn.Namespace, nn.Name)
-		return &errors.MissingDependencies{
-			MissingDeps: map[string][]string{"kafka": {qualifiedName}},
-		}
+		missingDeps := errors.MakeMissingDependencies(errors.MissingDependency{
+			Source:  "kafka",
+			App:     nn.Name,
```

Comment:
I believe nn.Name is the name of the KafkaTopic we tried to fetch, not the app's name.

Reply:
This one is more challenging. AFAICT there's no way for me to get a ClowdApp reference in this method without making changes more than one hop into the code. `appinterface.validateBrokerService` is called by `appinterface.NewAppInterface` which only has a reference to a Provider, which has no ClowdApp reference. And `appinterface.NewAppInterface` is `provider.GetKafka`, again with no ClowdApp reference. That's about as far as I'd be comfortable going to get this right. 

I'm going to make my next push without a resolution to this. I don't see a way to do it, so I'm open to suggestions.

Reply:
That said I did change the error to `Details: fmt.Sprintf("No kafka bootstrap service for KafkaTopic '%s' found in namespace '%s'", nn.Name, nn.Namespace),`